### PR TITLE
Refactor viewcontroller to use extension

### LINKED ISSUE #57: Create ViewController extensions
Please use extensions like stated in the Bible.

For example, FilterViewController is too big and should be split up. Very hard to read what is going on in that file.

### PR DESCRIPTION
**Pull request by jhseung**
Quick refactoring of some ViewController files to use extensions.

Fix #57

### CODE REVIEW

File: Fitness/Controllers/ClassDetailViewController.swift [hunk 1]
```diff
@@ -395,8 +395,10 @@ class ClassDetailViewController: UIViewController, UITableViewDelegate, UITableV
     @objc func favorite(){
         print("favorite")
```
Comment (AAAstorga) [inline Fitness/Controllers/ClassDetailViewController.swift]: Why do we do this? Let’s remove this print statement and just make this one line `@objc func favorite() {}`

File: Fitness/Controllers/FilterViewController.swift [hunk 2]
```diff
@@ -746,3 +533,224 @@ class FilterViewController: UIViewController, UICollectionViewDelegateFlowLayout
...
+            collectionView.selectItem(at: indexPath, animated: true, scrollPosition: .right)
+            cell.isSelected = true
+        }
+        return cell
+    }
+}
+
+//MARK: TableViewDataSource
+extension FilterViewController: UITableViewDataSource {
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
```
Comment (AAAstorga) [inline Fitness/Controllers/FilterViewController.swift]: Note for later, this method should probably get split up. Complicated currently. Make note please

File: Fitness/Controllers/FilterViewController.swift [hunk 3]
```diff
@@ -746,3 +533,224 @@ class FilterViewController: UIViewController, UICollectionViewDelegateFlowLayout
...
+    
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell = tableView.dequeueReusableCell(withIdentifier: DropdownViewCell.identifier, for: indexPath) as! DropdownViewCell
+        
+        if tableView == instructorDropdown{
+            if(indexPath.row < instructorDropdownData.titles.count){
+                cell.titleLabel.text = instructorDropdownData.titles[indexPath.row]
+                cell.id = instructorDropdownData.ids[indexPath.row]
+                
+                if(selectedInstructors.contains(cell.id)){
```
Comment (AAAstorga) [inline Fitness/Controllers/FilterViewController.swift]: Don’t need outer layer of `()`

File: Fitness/Controllers/FilterViewController.swift [hunk 4]
```diff
@@ -746,3 +533,224 @@ class FilterViewController: UIViewController, UICollectionViewDelegateFlowLayout
...
+    }
+}
+
+//MARK: TableViewDelegate
+extension FilterViewController: UITableViewDelegate {
+    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        let cell = tableView.cellForRow(at: indexPath) as! DropdownViewCell
+        var shouldAppend: Bool
+        
+        if(cell.checkBoxColoring.backgroundColor == .fitnessYellow){
```
Comment (AAAstorga) [inline Fitness/Controllers/FilterViewController.swift]: This if else check is weird and can be condensed. You can make `var shouldAppend = cell.checkBoxColoring.backgroundColor != .fitnessYellow` and do the same for the background color

File: Fitness/Controllers/FilterViewController.swift [hunk 5]
```diff
@@ -746,3 +533,224 @@ class FilterViewController: UIViewController, UICollectionViewDelegateFlowLayout
...
+            cell.checkBoxColoring.backgroundColor = .white
+            shouldAppend = false
+        }else{
+            cell.checkBoxColoring.backgroundColor = .fitnessYellow
+            shouldAppend = true
+        }
+        
+        if (tableView == classTypeDropdown){
```
Comment (AAAstorga) [inline Fitness/Controllers/FilterViewController.swift]: Remove ()

File: Fitness/Controllers/FilterViewController.swift [hunk 6]
```diff
@@ -746,3 +533,224 @@ class FilterViewController: UIViewController, UICollectionViewDelegateFlowLayout
...
+            if(shouldAppend){
```
Comment (AAAstorga) [inline Fitness/Controllers/FilterViewController.swift]: Remove ()

File: Fitness/Controllers/FilterViewController.swift [hunk 7]
```diff
@@ -746,3 +533,224 @@ class FilterViewController: UIViewController, UICollectionViewDelegateFlowLayout
...
+                selectedClasses.append(cell.id)
+            }else{
+                for i in 0..<selectedClasses.count{
+                    let id = selectedClasses[i]
+                    if(id == cell.id){
```
Comment (AAAstorga) [inline Fitness/Controllers/FilterViewController.swift]: And here

File: Fitness/Controllers/FilterViewController.swift [hunk 8]
```diff
@@ -746,3 +533,224 @@ class FilterViewController: UIViewController, UICollectionViewDelegateFlowLayout
...
+                        selectedClasses.remove(at: i)
+                        return
+                    }
+                }
+            }
+        }else{
+            if(shouldAppend){
```
Comment (AAAstorga) [inline Fitness/Controllers/FilterViewController.swift]: Here as well

File: Fitness/Controllers/FilterViewController.swift [hunk 9]
```diff
@@ -746,3 +533,224 @@ class FilterViewController: UIViewController, UICollectionViewDelegateFlowLayout
...
+        return 55
+    }
+    
+    func tableView(_ tableView: UITableView, heightForFooterInSection section: Int) -> CGFloat {
+        var height = 0
+        if tableView == classTypeDropdown {
+            switch classTypeDropdownData.dropStatus {
+            case .up:
+                height = 0
+            case .half:
```
Comment (AAAstorga) [inline Fitness/Controllers/FilterViewController.swift]: Combine cases `.half, .down`

File: Fitness/Controllers/FilterViewController.swift [hunk 10]
```diff
@@ -746,3 +533,224 @@ class FilterViewController: UIViewController, UICollectionViewDelegateFlowLayout
...
+                height = 32
+            case .down:
+                height = 32
+            default:
```
Comment (AAAstorga) [inline Fitness/Controllers/FilterViewController.swift]: Do we need default? Try and remove it

File: Fitness/Controllers/FilterViewController.swift [hunk 11]
```diff
@@ -746,3 +533,224 @@ class FilterViewController: UIViewController, UICollectionViewDelegateFlowLayout
...
+                height = 0
+            }
+        } else if tableView == instructorDropdown {
+            switch instructorDropdownData.dropStatus {
+            case .up:
```
Comment (AAAstorga) [inline Fitness/Controllers/FilterViewController.swift]: Condense cases here as well

File: Fitness/Controllers/FilterViewController.swift [hunk 12]
```diff
@@ -746,3 +533,224 @@ class FilterViewController: UIViewController, UICollectionViewDelegateFlowLayout
...
+    func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
+        return 32
+    }
+    
+    func tableView(_ tableView: UITableView, heightForHeaderInSection section: Int) -> CGFloat {
+        return 55
+    }
+    
+    func tableView(_ tableView: UITableView, heightForFooterInSection section: Int) -> CGFloat {
+        var height = 0
```
Comment (AAAstorga) [inline Fitness/Controllers/FilterViewController.swift]: `var height: CGFloat = 0.0`

File: Fitness/Controllers/FilterViewController.swift [hunk 13]
```diff
@@ -746,3 +533,224 @@ class FilterViewController: UIViewController, UICollectionViewDelegateFlowLayout
...
+                height = 0
+            case .half:
+                height = 32
+            case .down:
+                height = 32
+            default:
+                height = 0
+            }
+        }
+        return CGFloat(height)
```
Comment (AAAstorga) [inline Fitness/Controllers/FilterViewController.swift]: `return height`

File: Fitness/Controllers/FilterViewController.swift [hunk 14]
```diff
@@ -746,3 +533,224 @@ class FilterViewController: UIViewController, UICollectionViewDelegateFlowLayout
...
+    }
+    
+    func tableView(_ tableView: UITableView, viewForFooterInSection section: Int) -> UIView? {
+        let footer = tableView.dequeueReusableHeaderFooterView(withIdentifier: DropdownFooterView.identifier) as! DropdownFooterView
+        
+        if tableView == instructorDropdown {
+            let gesture = UITapGestureRecognizer(target: self, action: #selector(self.dropHideInstructors(sender:) ))
+            footer.addGestureRecognizer(gesture)
+            if (instructorDropdownData.dropStatus == .half) {
```
Comment (AAAstorga) [inline Fitness/Controllers/FilterViewController.swift]: Maybe swiftlint will fix this, but yeah I’m going to stop leaving comments for removing `()` but please do it everywhere it exists. Thanks!

File: Fitness/Controllers/GymDetailViewController.swift [hunk 15]
```diff
@@ -364,26 +364,70 @@ class GymDetailViewController: UIViewController, UITableViewDelegate, UITableVie
...
+        setupConstraints()
+    }
+    
+    //MARK: - BUTTON METHODS
+    @objc func back() {
+        navigationController!.popViewController(animated: true)
+    }
+    
+    @objc func favorite(){
+        print("favorite")
```
Comment (AAAstorga) [inline Fitness/Controllers/GymDetailViewController.swift]: Remove print and condense into one line

File: Fitness/Controllers/GymDetailViewController.swift [hunk 16]
```diff
@@ -460,73 +504,35 @@ class GymDetailViewController: UIViewController, UITableViewDelegate, UITableVie
...
-        }
-    }
-    
+}
+
+//MARK: TableViewDelegate
+extension GymDetailViewController: UITableViewDelegate {
     func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
-        if(tableView == hoursTableView){
+        if (tableView == hoursTableView) {
```
Comment (AAAstorga) [inline Fitness/Controllers/GymDetailViewController.swift]: You can combine this. Like `return indexPath.row == 5 ? 19 : 27

File: Fitness/Controllers/HomeController.swift [hunk 17]
```diff
@@ -159,7 +158,15 @@ class HomeController: UIViewController, UITableViewDataSource, UITableViewDelega
...
     
-    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
-        return 1
+    func tableView(_ tableView: UITableView, heightForHeaderInSection section: Int) -> CGFloat {
+        
+        switch sections[section] {
+        case .allGyms:
+            return 155
+        case .todaysClasses:
+            return 51
```
Comment (AAAstorga) [inline Fitness/Controllers/HomeController.swift]: Condense same enum cases

**Comment by jhseung** [comment]
Leaving parentheses fixes in there – if SiwftLint doesn't fix them, will go through and manually fix them later.

**Pull request merged**